pkg/app/instances/docker.go

```diff
@@ -117,16 +123,78 @@ func (m *DockerInstanceManager) DeleteHost(zone string, _ accounts.User, host st
 		return nil, fmt.Errorf("Failed to remove docker container: %w", err)
 	}
 	return &apiv1.Operation{
-		Name: host,
+		Name: EncodeOperationName(DeleteHostOPType, host),
 		Done: true,
 	}, nil
 }
 
-func (m *DockerInstanceManager) WaitOperation(zone string, _ accounts.User, _ string) (any, error) {
+func EncodeOperationName(opType string, host string) string {
+	return opType + "_" + host
+}
+
+func DecodeOperationName(name string) (string, string, error) {
+	words := strings.SplitN(name, "_", 2)
+	if len(words) == 2 {
+		return words[0], words[1], nil
+	} else {
+		return "", "", errors.NewBadRequestError(fmt.Sprintf("cannot parse operation from %q", name), nil)
+	}
+}
+
+func (m *DockerInstanceManager) waitCreateHostOperation(host string) (*apiv1.HostInstance, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 3*time.Minute)
+	defer cancel()
```

Comment:
If every API we use is sync, we don't need timeout, just check and then return the result.

Reply:
In our usages, I think your opinion is right. However because of definition of this API, I think it's still better to have timeout mechanism if it doesn't harm any correctness or performance issue.

But I prefer to follow your opinion if you still want, since it has no actual usage. HDYT? @ikicha

Reply:
I am just worried that it needs to wait until timeout for an error case even though there is no chance to be recovered